core/commands/pull.py

```diff
@@ -6,22 +6,41 @@
 from ..ui_mixins.quick_panel import show_branch_panel
 
 
-class GsPullCommand(WindowCommand, GitCommand):
+class GsPull(WindowCommand, GitCommand):
+    """
+    Pull from remote tracking branch if it is found. Otherwise, use GsPullFromBranchCommand.
+    """
+
+    def run(self):
+        sublime.set_timeout_async(self.run_async)
+
+    def run_async(self):
+        # honor the `pull.rebase` config implictly
+        rebase = self.git("config", "pull.rebase", throw_on_stderr=False) or False
+        if rebase and rebase.strip() == "true":
+            rebase = True
+        upstream = self.get_upstream_for_active_branch()
+        if upstream:
+            remote, remote_branch = upstream.split("/", 1)
+            self.pull(remote=remote, remote_branch=remote_branch, rebase=rebase)
```

Comment:
This is only a subset of what `GsPullFromBranchCommand.do_pull` does.

Instead of calling `pull()` here directly, can we perhaps extend the more specific class to accept remote/branch and:

```python
self.window.run_command("gs_pull_from_branch", {"rebase": rebase, "remote": remote, "remote_branch": remote_branch})
```

Alternatively, we can refactor these commands to both define `def do_pull(self, remote, remote_branch):`; either by inheritance or with a mixin...

Reply:
added a `GsPullBase` class which is similar to `GsPushBase`